src/com/sap/piper/ConfigurationMerger.groovy

```diff
@@ -20,6 +38,15 @@ class ConfigurationMerger {
         return merged
     }
 
+    @NonCPS
+    def static mergeDeepStructure(Map parameters, Map parameterKeys, Map configuration, Map configurationKeys, Map defaults=[:]){
```

Comment:
I would like to suggest not to have a mergeDeepStructure method exposed to outside class, but doing the deep merge inside the merge as we have it today. This means: we always perform a deep merge. IMO this is fine.

Reply:
Therefore we would need to change the marging to work with a key map instead of a key list, would this be ok?

Reply:
Yes, in this case the keys are a map, since we specifiy sub-keys for each keys, in a recursive manner, of course.

Reply:
With this we have the new merge(Map parameters, Map parameterKeys, Map configuration, List configurationKeys, Map defaults).
For the time we need to be backward compatible there would be also the old method(Map parameters, List parametersKeys, Map configuration, List configurationKeys, Map defaults). IMO this method should redirect the the new methods, marshalling the list into the keyset with null values. Of course deep-merging with this redirection does not work. This needs to be clearly outlined in a log message.

Reply:
`configurationKeys` should also be a map